src/modules/sensors/vehicle_acceleration/VehicleAcceleration.cpp

```diff
@@ -109,39 +110,71 @@ VehicleAcceleration::SensorCorrectionsUpdate(bool force)
 		sensor_correction_s corrections{};
 		_sensor_correction_sub.copy(&corrections);
 
-		// TODO: should be checking device ID
-		if (_selected_sensor == 0) {
+		// find in sensor corrections
+		if ((_sensor_correction_index < 0)
+		    || (corrections.accel_device_id[_sensor_correction_index] != _selected_sensor_device_id)) {
+
+			for (int correction_index = 0; correction_index < MAX_SENSOR_COUNT; correction_index++) {
+				if (corrections.accel_device_id[correction_index] == _selected_sensor_device_id) {
+					_sensor_correction_index = correction_index;
```

Comment:
I'm not sure I understand this. Is `_sensor_correction_index` valid for the remainder? And if so, we only ever correct one of the sensors?

Reply:
>  Is _sensor_correction_index valid for the remainder

No, unless there's a bug I am not seeing. `SensorSelectionUpdate()` is called everytime thru the loop in `Run()` and it will perform the check for `data.device_id == selection.accel_device_id` and if this fails it calls `SensorCorrectionsUpdate` which will update `_sensor_correction_index`. It's kinda tangled and messy and I don't particularly like it, but as far as I am aware this "works" (if it aint broke don't fix it?).